execution-plan-traits/src/address.rs

```diff
@@ -34,6 +34,49 @@ impl Address {
         self.0 += size;
         old
     }
+
+    /// Returns an iterator to safely get the next available address.
+    /// Mutates self so subsequent calls to allocate() move the
+    /// "address counter" forward.
+    pub fn allocate(&mut self, n: usize) -> AddressIterator {
```

Comment:
Neither this function nor `AddressIterator` are used anywhere -- is this meant to be used from Grackle?

Reply:
Good catch, I think I eventually factored it out - any ideas why `clippy` wouldn't catch something like this? I'd like to avoid it in the future. I thought Rust itself emitted a "deadcode" error...